lib/gollum/views/helpers.rb

```diff
@@ -2,31 +2,63 @@
 
 module Precious
   module Views
+    module RouteHelpers
+      ROUTES = {
+        'gollum' => {
+          :last_commit_info => 'last-commit-info',
+          :latest_changes => 'latest_changes',
+          :upload_file => 'uploadFile',
+          :create => 'create',
+          :delete => 'delete',
+          :edit => 'edit',
+          :pages => 'pages',
+          :history => 'history',
+          :rename => 'rename',
+          :preview => 'preview',
+          :compare => 'compare',
+          :search => 'search'
+        }
+      }
+
+      def self.parse_routes(routes, prefix = '')
+        routes.each do |name, path|
+          if path.respond_to?(:keys) then
+            self.parse_routes(path, "#{prefix}/#{name}")
+          else
+            define_method :"#{name.to_s}_path" do
+              "#{base_url}/#{prefix}/#{path}".gsub(/\/{2,}/, '/')
```

Comment:
Is there anyway to make clearer what the gsub does here?

Also, could `gsub!` be appropriate?

Reply:
I think `gsub!` might end up modifying `routes`, but I think I can get rid of the substitution entirely by only conditionally including `prefix`.

Reply:
That's not going to work unfortunately, because `base_url` is another source of double slashes (in case it is `'/'`). Do you have a proposal for a better readable `gsub`? Otherwise I could just add a comment.

Reply:
A comment would be sufficient. 👍